src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,387 @@
+  description?: string;
+  orderHint?: string;
+}
+
+class PlannerTaskAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.TASK_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new Microsoft Planner Task';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planName = typeof args.options.planName !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    telemetryProps.bucketId = typeof args.options.bucketId !== 'undefined';
+    telemetryProps.bucketName = typeof args.options.bucketName !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.startDateTime = typeof args.options.startDateTime !== 'undefined';
+    telemetryProps.dueDateTime = typeof args.options.dueDateTime !== 'undefined';
+    telemetryProps.percentComplete = typeof args.options.percentComplete !== 'undefined';
+    telemetryProps.assignedToUserIds = typeof args.options.assignedToUserIds !== 'undefined';
+    telemetryProps.assignedToUserNames = typeof args.options.assignedToUserNames !== 'undefined';
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    telemetryProps.orderHint = typeof args.options.orderHint !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['title', 'planId', 'bucketId'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getPlanId(args)
+      .then((planId: string) => 
+        this.getBucketId(args, planId).then(async (bucketId: string) => {
+          const assignments: { [userId: string]: any; } = await this.generateUserAssignments(args, logger);
+
+          const requestOptions: any = {
+            url: `${this.resource}/v1.0/planner/tasks`,
+            headers: {
+              'accept': 'application/json;odata.metadata=none'
+            },
+            responseType: 'json',
+            data: {
+              planId: planId,
+              bucketId: bucketId,
+              title: args.options.title,
+              startDateTime: args.options.startDateTime,
+              dueDateTime: args.options.dueDateTime,
+              percentComplete: args.options.percentComplete,
+              assignments: assignments,
+              orderHint: args.options.orderHint
+            }
+          };
+
+          return request
+            .post(requestOptions)
+            .then(async (newTask: any) => {
+              const taskId = newTask['id'];
+
+              if (args.options.description) {
+                const etag = await this.getTaskDetails(taskId);
```

Comment:
Since we just created the task, couldn't we use `*` as etag instead of retrieving it?

Reply:
I don't understand the etag logic 100% but when I pass a `*` as etag, I get the following error `Error: The given key was not present in the dictionary.`

Reply:
Etag is meant to ensure item integrity. In other words, before you update an item, you need to prove that you know what item you're updating and that its contents haven't changed between when you retrieved it and when you're updating it. Passing `*` as etag basically tells the API to ignore the integrity check and overwrite whatever contents are there. It's odd that it's not working here, but if that's the case, then let's keep the current code. One more thing that we could check is if we're not getting etag with the server response after creating the item. If we do, then it would save us an extra roundtrip to the server.

Reply:
Oh wow. That's a great explanation, thanks. 

Problem with the etag from our task creation is that it isn't the same like task details. Meaning if I use the etag from task creation, I again get the error `Error: The given key was not present in the dictionary.`

Reply:
Ah right! I forgot that task and task details are two different things. You're right 👍